tests/e2e/components_test.go

```diff
@@ -0,0 +1,280 @@
+package e2e_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/common"
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/matchers/jq"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/utils/test/testf"
+
+	. "github.com/onsi/gomega"
+)
+
+type ComponentTestCtx struct {
+	*testf.TestContext
+
+	GVK                  schema.GroupVersionKind
+	DSCName              types.NamespacedName
+	DSCIName             types.NamespacedName
+	ApplicationNamespace string
+}
+
+func NewComponentTestCtx(object common.PlatformObject) (*ComponentTestCtx, error) {
+	tcf, err := testf.NewTestContext(
+		testf.WithTOptions(
+			testf.WithEventuallyTimeout(2*time.Minute),
+			testf.WithEventuallyPollingInterval(1*time.Second),
+		),
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	ogvk, err := resources.GetGroupVersionKindForObject(tcf.Scheme(), object)
+	if err != nil {
+		return nil, err
+	}
+
+	dsciList := dsciv1.DSCInitializationList{}
+	if err := tcf.Client().List(tcf.Context(), &dsciList); err != nil {
+		return nil, err
+	}
+
+	if len(dsciList.Items) != 1 {
+		return nil, fmt.Errorf("failure looking up DSCInitialization, expected=1, found=%d", len(dsciList.Items))
+	}
+
+	dscList := dscv1.DataScienceClusterList{}
+	if err := tcf.Client().List(tcf.Context(), &dscList); err != nil {
+		return nil, err
+	}
+
+	if len(dscList.Items) != 1 {
+		return nil, fmt.Errorf("failure looking up DataScienceCluster, expected=1, found=%d", len(dscList.Items))
+	}
+
+	componentCtx := ComponentTestCtx{
+		TestContext:          tcf,
+		GVK:                  ogvk,
+		DSCName:              client.ObjectKeyFromObject(&dscList.Items[0]),
+		DSCIName:             client.ObjectKeyFromObject(&dsciList.Items[0]),
+		ApplicationNamespace: dsciList.Items[0].Spec.ApplicationsNamespace,
+	}
+
+	return &componentCtx, nil
+}
+
+func (c *ComponentTestCtx) ValidateComponentEnabled(t *testing.T) {
+	g := c.NewWithT(t)
+
+	g.Update(
+		gvk.DataScienceCluster,
+		c.DSCName,
+		testf.Transform(`.spec.components.%s.managementState = "%s"`, strings.ToLower(c.GVK.Kind), operatorv1.Managed),
+	).Eventually().WithTimeout(30 * time.Second).WithPolling(1 * time.Second).Should(
```

Comment:
I'm wondering why default timings are not applicable in such cases?

Reply:
done